microservices-connector/cmd/router/main.go

```diff
@@ -94,6 +111,221 @@ type ReadCloser struct {
 	*bytes.Reader
 }
 
+var (
+	firstTokenLatencyMeasure metric.Float64Histogram
+	nextTokenLatencyMeasure  metric.Float64Histogram
+	allTokenLatencyMeasure   metric.Float64Histogram
+	pipelineLatencyMeasure   metric.Float64Histogram
+	stepLatencyMeasure       metric.Float64Histogram
+)
+
+func initMeter() {
+	// The exporter embeds a default OpenTelemetry Reader and
+	// implements prometheus.Collector, allowing it to be used as
+	// both a Reader and Collector.
+	exporter, err := prometheus.New()
+	if err != nil {
+		log.Error(err, "metrics: cannot init prometheus collector")
+	}
+	provider := sdkmetric.NewMeterProvider(sdkmetric.WithReader(exporter))
+	otel.SetMeterProvider(provider)
+
+	// ppalucki: Own metrics defintion bellow
+	const meterName = "entrag-telemetry"
+	meter := provider.Meter(meterName)
+
+	firstTokenLatencyMeasure, err = meter.Float64Histogram(
+		"llm.first.token.latency",
+		metric.WithUnit("ms"),
+		metric.WithDescription("Measures the duration of first token generation."),
+		api.WithExplicitBucketBoundaries(1, 64, 128, 256, 512, 1024, 2048, 4096, 8192, 16364),
+	)
+	if err != nil {
+		log.Error(err, "metrics: cannot register first token histogram measure")
+	}
+	nextTokenLatencyMeasure, err = meter.Float64Histogram(
+		"llm.next.token.latency",
+		metric.WithUnit("ms"),
+		metric.WithDescription("Measures the duration of generating all but first tokens."),
+		api.WithExplicitBucketBoundaries(1, 64, 128, 256, 512, 1024, 2048, 4096, 8192, 16364),
+	)
+	if err != nil {
+		log.Error(err, "metrics: cannot register next token histogram measure")
```

Comment:
Similar improvement could be applied to all metric descriptions / errors:
```suggestion
		metric.WithDescription("Duration of generating all but the first tokens."),
		api.WithExplicitBucketBoundaries(1, 64, 128, 256, 512, 1024, 2048, 4096, 8192, 16364),
	)
	if err != nil {
		log.Error(err, "metrics: cannot register next token duration histogram")
```